### ISSUE TITLE
Deprecate `--ignore-taint` , use `--startup-taint` instead

### ISSUE THREAD

**Issue by himanshu-kun** [gardener/autoscaler#259]
**What would you like to be added**:

`--ignore-taint` has been deprecated in upstream autoscaler by the PR https://github.com/kubernetes/autoscaler/pull/6132 . New taint [`--startup-taint`](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-cluster-autoscaler-treat-nodes-with-statusstartupignore-taints) has been added in its place.. Since we already consume `--ignore-taint` in [shoot Yaml ](https://github.com/gardener/gardener/blob/4a7ec00a52c9af5a995e244d1d073bd12a3e89ab/example/90-shoot.yaml#L289-L291), we should adapt before this taint is dropped by upstream.

**Why is this needed**:
To shift to new taints in time.

**Comment by takoverflow** [comment]
From `90-shoot.yaml`: Looks like this has been [updated](https://github.com/gardener/gardener/blob/0ac4022d6a12ffece3400c68c217955ef4925942/example/90-shoot.yaml#L332) and can be closed @aaronfern 

Related PR: https://github.com/gardener/gardener/pull/10858

**Comment by aaronfern** [comment]
Not really, the older `ignore-taint` flag is still used (ref: https://github.com/gardener/gardener/blob/0ac4022d6a12ffece3400c68c217955ef4925942/pkg/component/autoscaling/clusterautoscaler/cluster_autoscaler.go#L464)
We should close this only after we deprecate the older flag as well
